addons/l10n_it_edi/models/account_invoice.py

```diff
@@ -211,10 +209,10 @@ def format_phone(number):
             return False
 
         def get_vat_number(vat):
-            return vat[2:].replace(' ', '')
+            return vat[2:].replace(' ', '') if vat else '0'
 
         def get_vat_country(vat):
-            return vat[:2].upper()
+            return vat[:2].upper() if vat else '0'
```

Comment:
Should not we return False and provide a fallback?

Reply:
@jco-odoo I checked again found `get_vat_country(vat)/get_vat_number(vat)` will always have value for `vat`(except for one corner case) as it is mainly called for
1) invoice's <a href="https://github.com/odoo/odoo/blob/master/addons/l10n_it_edi/data/invoice_it_template.xml#L72">company_id</a>
2) <a href="https://github.com/odoo/odoo/blob/master/addons/l10n_it_edi/data/invoice_it_template.xml#L97">l10n_it_tax_representative_partner_id</a> of company(for which we have a <a href="https://github.com/odoo/odoo/blob/master/addons/l10n_it_edi/models/res_company.py#L107">python constraint</a> that doesn't allow a partner without VAT)
3) for <a href="https://github.com/odoo/odoo/blob/master/addons/l10n_it_edi/data/invoice_it_template.xml#L111">commercial_partner_id</a> of invoice <a href="https://github.com/odoo/odoo/blob/master/addons/l10n_it_edi/data/invoice_it_template.xml#L110">only if</a> it has vat number

So, unless someone mistakenly remove VAT from the `l10n_it_tax_representative_partner_id` after setting it in company(which practically is unlikely), IMO we do not need to have fallback. Instead I suggest to add check for this in `_check_before_xml_exporting` like,
```
if seller.l10n_it_has_tax_representative and not seller.l10n_it_tax_representative_partner_id.vat:
    raise UserError(_("Tax representative partner %s of %s must have a tax number.") % (seller.l10n_it_tax_representative_partner_id.display_name, seller.display_name))

```
What do you say ?

Reply:
@rgo-odoo ok for me

Reply:
@jco-odoo done